lambda-api/src/main/java/com/formkiq/stacks/api/handler/DocumentsCompressRequestHandler.java

```diff
@@ -0,0 +1,112 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.formkiq.stacks.api.handler;
+
+import com.amazonaws.services.lambda.runtime.LambdaLogger;
+import com.formkiq.aws.dynamodb.DynamicObject;
+import com.formkiq.aws.s3.PresignGetUrlConfig;
+import com.formkiq.aws.s3.S3Service;
+import com.formkiq.aws.services.lambda.ApiAuthorization;
+import com.formkiq.aws.services.lambda.ApiGatewayRequestHandler;
+import com.formkiq.aws.services.lambda.ApiGatewayRequestEvent;
+import com.formkiq.aws.services.lambda.ApiGatewayRequestEventUtil;
+import com.formkiq.aws.services.lambda.ApiRequestHandlerResponse;
+import com.formkiq.aws.services.lambda.ApiMapResponse;
+import com.formkiq.module.lambdaservices.AwsServiceCache;
+
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Map;
+import java.util.UUID;
+
+import static com.formkiq.aws.services.lambda.ApiResponseStatus.SC_CREATED;
+import static java.util.Map.entry;
+
+/** {@link ApiGatewayRequestHandler} for "/documents/compress". */
+public class DocumentsCompressRequestHandler
+    implements ApiGatewayRequestHandler, ApiGatewayRequestEventUtil {
+
+  @Override
+  public String getRequestUrl() {
+    return "/documents/compress";
+  }
+
+  @Override
+  public ApiRequestHandlerResponse post(final LambdaLogger logger,
+      final ApiGatewayRequestEvent event, final ApiAuthorization authorization,
+      final AwsServiceCache awsServices) throws Exception {
+    DynamicObject requestBodyObject = fromBodyToDynamicObject(event);
+
+    final String siteId = authorization.siteId();
+    final String compressionId = UUID.randomUUID().toString();
+    final String compressionTaskS3Key = getS3Key(siteId, compressionId, false);
+    S3Service s3 = awsServices.getExtension(S3Service.class);
+    final String stagingBucket = awsServices.environment("STAGE_DOCUMENTS_S3_BUCKET");
+    final String downloadUrl =
+        getArchiveDownloadUrl(s3, stagingBucket, getS3Key(siteId, compressionId, true));
+    final DynamicObject taskObject =
+        getS3TaskObject(requestBodyObject, siteId, compressionId, downloadUrl);
+
+    putObjectToStaging(s3, stagingBucket, compressionTaskS3Key, GSON.toJson(taskObject));
+    ApiMapResponse response =
+        new ApiMapResponse(Map.of("compressionId", compressionId, "downloadUrl", downloadUrl));
```

Comment:
I think "documentId" is more constant with the existing API than "compressionId"

Reply:
Done, changed to documentId.